src/main/kotlin/blackJack/domain/card/Cards.kt

```diff
@@ -15,22 +18,29 @@ class Cards(private val cards: List<Card>) : List<Card> by cards {
         require(card !in cards) { DUPLICATE_ERROR }
     }
 
-    fun sumCards(): Int =
-        cards.sortedByDescending { it.denomination.orderBy }.fold(0) { acc: Int, card: Card ->
-            if (card.denomination == Denomination.ACE) {
-                val aceScore = card.denomination.score(acc)
-                acc + card.denomination.score(acc + aceScore)
-            } else {
-                acc + card.denomination.score(acc)
-            }
-        }.let { sum ->
-            if (sum > 21 && cards.map { it.denomination }.contains(Denomination.ACE)) {
-                cards.sumOf { it.denomination.score(sum) }
-            } else {
-                sum
-            }
+    fun sumCards(): Int {
+        val total = cards.sumOf { it.denomination.number }
+        return sumBonus(total)
+    }
+
+    private fun sumBonus(total: Int): Int =
+        if (isAce()) {
+            addAceBonus(total)
+        } else {
+            total
         }
 
+    private fun isAce() = cards.map { it.denomination }.contains(Denomination.ACE)
```

Comment:
[any](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/any.html)를 활용하면 아래와 같이 만들 수 있겠네여!
```suggestion
    private fun isAce() = cards.any { it.denomination === Denomination.ACE }
```